### PR TITLE
Add audformat.utils.map_file_path()

### PR DESCRIPTION
**Pull request by hagenw**
Adds a helper function to map file path in index by callable.

image

### CODE REVIEW

**Comment by hagenw** [comment]
We are mainly using `.map(os.path.normpath)` here, but need to apply it differently for filewise and segmented index.
An alternative to `audformat.utils.norm_file_path()` would be to implement a generic `audformat.utils.map_file_path(index, callable)`. If we need to norm the path we could then do `audformat.utils.map_file_path(index, os.path.normpath)`.

**Comment by frankenjoe** [comment]
> audformat.utils.map_file_path(index, callable)

I'm in favor of this solution.

**Comment by hagenw** [comment]
> > audformat.utils.map_file_path(index, callable)
> 
> I'm in favor of this solution.

Changed it accordingly.

File: tests/test_utils.py [hunk 1]
```diff
@@ -913,6 +913,36 @@ def test_join_schemes():
...
+            os.path.normpath,
+            audformat.segmented_index(['a/f1'], [0], [1]),
+            audformat.segmented_index(['a\\f1'], [0], [1]),
+        ),
+    ]
+)
+def test_map_file_path(index, func, expected_index, expected_index_windows):
+    mapped_index = audformat.utils.map_file_path(index, func)
+    if os.name == 'nt':
+        expected_index = expected_index_windows
```
[the remark this answers is not in the document]
Comment (hagenw) [reply tests/test_utils.py:942]: No, it is not automatically changed to `\` on Windows. I'm testing `map_file_path()` with the callable `os.path.normpath()` here, which converts `/` to `\` on Windows.

As we need this behavior, I thought I use it for the test here already. Or do you think the test is too complicated to understand at the moment?

[the remark this answers is not in the document]
Comment (frankenjoe) [reply tests/test_utils.py:942]: > No, it is not automatically changed to `\` on Windows. I'm testing `map_file_path()` with the callable `os.path.normpath()` here, which converts `/` to `\` on Windows.

Yes, I know it depends on the function that you pass to to `map_file_path()` and that's why I was thinking we should maybe - after calling that function - always replace `\` with `/` before we return the index. Or we don't want to do it by default add an argument that ensures portability.

[the remark this answers is not in the document]
Comment (hagenw) [reply tests/test_utils.py:942]: The mean reason why I introduce `map_file_path()` here is that we can return `\` on Windows. As I think for `audb.load(..., full_path=True)` we need to return the correct path.

I would not add anything on top pf the current behavior and leave it to the user how to use it.

[the remark this answers is not in the document]
Comment (frankenjoe) [reply tests/test_utils.py:942]: Ok, I see.

File: audformat/core/utils.py [hunk 2]
```diff
@@ -543,6 +543,48 @@ def join_schemes(
         db.schemes[scheme_id].replace_labels(labels)
 
 
+def map_file_path(
+        index: pd.Index,
+        func: typing.Callable,
```
Comment (frankenjoe) [inline audformat/core/utils.py:548]:
```suggestion
        func: typing.Callable[[str], str],
```

**Pull request merged**